nc/views.py

```diff
@@ -40,6 +41,37 @@ def search(request):
 class AgencyList(ListView):
     model = Agency
 
+    def get_context_data(self, **kwargs):
+        context = super(AgencyList, self).get_context_data(**kwargs)
+
+        if self.request.method == 'GET' and self.request.GET:
+            form = forms.AgencySearchForm(self.request.GET)
```

Comment:
I'm curious: Did you try adding FormMixin to AgencyList()?  That seemingly would be the natural CBV way to implement the form handling.  I don't know if there's something about this particular situation that would make it troublesome.